xdsl/dialects/riscv_func.py

```diff
@@ -48,9 +51,102 @@ def verify_(self):
             )
 
 
+@irdl_op_definition
+class CallOp(IRDLOperation):
+    """RISC-V function call operation"""
+
+    name = "riscv_func.call"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    func_name: OpAttr[StringAttr]
+    result: Annotated[OptOpResult, riscv.RegisterType]
```

Comment:
I'm pretty sure that we want to have `VarResult` here instead. It's common to return values in multiple registers to prevent values spilling to the stack (e.g. when returning a 64 bit value, or a struct)